docs/package.nix

```
@@ -0,0 +1,138 @@
+                            scrubDerivations wholeName value
+                            // optionalAttrs (isDerivation value) {
+                              inherit (value) drvPath;
+                              outPath = "\${${wholeName}}";
+                            }
+                          else value
+                      )
+                      pkgSet;
+                  in {
+                    _module = {
+                      check = false;
+                      args.pkgs = mkForce (scrubDerivations "pkgs" pkgs);
+                    };
+                  }
+                )
+                # avoid having `_module.args` in the documentation
+                {
+                  options = {
+                    _module.args = mkOption {
+                      internal = true;
+                    };
+                  };
+                }
+              ];
+          })
+        )
+        options
+        ;
+
+      transformOptions = opt:
+        opt
+        // {
+          declarations =
+            map (
+              decl:
+                if hasPrefix (toString ../.) (toString decl)
+                then
+                  pipe decl [
+                    toString
+                    (removePrefix (toString ../.))
+                    (removePrefix "/")
+                    (x: {
+                      url = "https://github.com/feel-co/hjem/blob/main/${x}";
+                      name = "<hjem/${x}>";
+                    })
+                  ]
+                else if decl == "lib/modules.nix"
+                then {
+                  url = "https://github.com/NixOS/nixpkgs/blob/master/${decl}";
+                  name = "<nixpkgs/lib/modules.nix>";
+                }
+                else decl
+            )
+            opt.declarations;
+        };
+    })
+    .optionsJSON;
+
+  hjemDocsWeb =
+    pkgs.runCommandLocal "hjem-docs" {
+      nativeBuildInputs = [ndg];
+    } ''
+      mkdir -p $out/share/doc
+
+      # Copy the markdown sources to be processed by ndg
+      cp -rvf ${./inputs} ./inputs
+
+      ndg --verbose html \
+        --jobs $NIX_BUILD_CORES --title "Hjem" \
+        --module-options ${configJSON}/share/doc/nixos/options.json \
+        --manpage-urls ${pkgs.path}/doc/manpage-urls.json \
+        --options-depth 3 \
+        --generate-search true \
+        --highlight-code true \
+        --input-dir ./inputs \
+        --output-dir "$out/share/doc"
+    '';
+in {
+  html = hjemDocsWeb;
+  options.json =
+    pkgs.runCommand "options.json" {
+      meta.description = "List of Hjem options in JSON format";
+    } ''
+      mkdir -p $out/{share/doc,nix-support}
+
+      cp -a ${configJSON.optionsJSON}/share/doc/nixos $out/share/doc/hjem
+
+      substitute \
+        ${configJSON.optionsJSON}/nix-support/hydra-build-products \
+        $out/nix-support/hydra-build-products \
+          --replace '${configJSON.optionsJSON}/share/doc/nixos' "$out/share/doc/hjem"
+    '';
```

```suggestion
      cp -a ${configJSON}/share/doc/nixos $out/share/doc/hjem

      substitute \
        ${configJSON}/nix-support/hydra-build-products \
        $out/nix-support/hydra-build-products \
          --replace '${configJSON}/share/doc/nixos' "$out/share/doc/hjem"
    '';
```

`configJSON` already has `.optionsJSON` at the end, so this was effectively `.optionsJSON.optionsJSON`